lib/internal/stream_base_commons.js

```diff
@@ -80,8 +80,7 @@ function onWriteComplete(status) {
   }
 
   if (status < 0) {
-    const ex = errnoException(status, 'write', this.error);
-    stream.destroy(ex, this.callback);
+    this.callback(errnoException(status, 'write', this.error));
```

Comment:
Aren't we missing a `if (typeof this.callback === 'function')` like above?

Reply:
I don't think that check should be required... callback should never be falsy. Added it just in case anyway.